zanata-common-api/src/main/java/org/zanata/rest/service/ProjectVersionResource.java

```diff
@@ -0,0 +1,267 @@
+     */
+    @GET
+    @Produces({ MediaTypes.APPLICATION_ZANATA_VERSION_LOCALES_JSON,
+            MediaType.APPLICATION_JSON })
+    @Path(VERSION_SLUG_TEMPLATE + "/locales")
+    @TypeHint(LocaleDetails[].class)
+    public Response getLocales(@PathParam("projectSlug") String projectSlug,
+        @PathParam("versionSlug") String versionSlug);
+
+    /**
+     * Retrieve the List of Documents (Resources) belongs to a Project version.
+     *
+     * @param projectSlug
+     *            Project identifier
+     * @param versionSlug
+     *            Project version identifier
+     *
+     * @return The following response status codes will be returned from this
+     *         operation:<br>
+     *         OK(200) - Response with a list of documents wrapped in a
+     *         "resources" element. Each child element will be a
+     *         "resource-meta". <br>
+     *         NOT FOUND(404) - If a Project iteration could not be found with
+     *         the given parameters.<br>
+     *         INTERNAL SERVER ERROR(500) - If there is an unexpected error in
+     *         the server while performing this operation.
+     */
+    @GET
+    @Produces({ MediaType.APPLICATION_JSON })
+    @Path(VERSION_SLUG_TEMPLATE + "/docs")
+    @TypeHint(ResourceMeta[].class)
+    public Response getDocuments(@PathParam("projectSlug") String projectSlug,
+        @PathParam("versionSlug") String versionSlug);
+
+    /**
+     * Retrieves a list translation unit with status in a document.
+     *
+     * @param projectSlug
+     *            Project identifier
+     * @param versionSlug
+     *            Project version identifier
+     * @param docId
+     *            The document identifier. Some document ids could have forward
+     *            slashes ('/') in them which would cause conflicts with the
+     *            browser's own url interpreter. For this reason, the supplied
+     *            id must have all its '/' characters replaced with commas
+     *            (',').
+     * @param localeId
+     *            target locale
+     *
+     * @return The following response status codes will be returned from this
+     *         operation:<br>
+     *         OK(200) - Response containing a full list of locales. <br>
+     *         NOT FOUND(404) - If a document or locale could not be found for
+     *         the given parameters.<br>
+     *         INTERNAL SERVER ERROR(500) - If there is an unexpected error in
+     *         the server while performing this operation.
+     */
+    @GET
+    @Produces({ MediaTypes.APPLICATION_ZANATA_TRANS_UNIT_RESOURCE_JSON,
+            MediaType.APPLICATION_JSON })
+    @Path(VERSION_SLUG_TEMPLATE + "/doc/{docId}/status/{localeId}")
+    @TypeHint(LocaleDetails[].class)
+    public Response getTransUnitStatus(
+        @PathParam("projectSlug") String projectSlug,
+        @PathParam("versionSlug") String versionSlug,
+        @PathParam("docId") String docId,
```

Comment:
What about space in doc path? or ```C:\My Documents\some file.txt```? From memory at some point we were talking about migrating docId from path to param to avoid all this mess. LocaleId may be ```kw@kkcor``` which is quite ugly after encoding in path. If this is a new API we may take the opportunity to refactor this.